client/js/render.js

```diff
@@ -158,23 +163,59 @@ function renderChannelUsers(data) {
 	}
 }
 
-function renderNetworks(data) {
+function renderNetworks(data, singleNetwork) {
```

Comment:
I'm not sure to see what `singleNetwork` is for, could you explain?

Reply:
Gotta do different logic when rendering channels on init when it's reconnecting, so that it won't render duplicate channels.

Reply:
I'm still not a fan of the implementation though. It's essentially 2 different functions in 1 with a boolean to switch implementation so that's why I think it reads a bit bizarre.

I'll try to think of other ways as I continue with the review, but on the top of my head, what do you think of either: 2 different functions, calling the correct one when needed (with potentially a common function/helper for shared logic); or a `renderNetworks` that calls multiple times a `renderNetwork` (what we do when building channels and messages), calling the former one on init and the latter one when it's reconnecting?

Reply:
That will effectively be duplicate code, and it doesn't solve the problem.